### PR TITLE
Upgrade rust toolchain

### PR DESCRIPTION
**Pull request by bitfriend**
Will fix https://github.com/acterglobal/a3/issues/1302

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ jobs:
       - name: Install supported toolchain
         uses: actions-rs/toolchain@v1
         with:
-          toolchain: nightly-2023-04-10
+          toolchain: nightly-2024-01-19
```
Comment (gnunicorn) [inline .github/workflows/deploy.yml:22]: let's pin to no specific version but `nightly` directly please

File: .github/workflows/styles.yml [hunk 2]
```diff
@@ -20,7 +20,7 @@ jobs:
       - name: Install specific toolchain
         uses: actions-rs/toolchain@v1
         with:
-          toolchain: nightly-2023-04-10
+          toolchain: nightly-2024-01-19
```
Comment (gnunicorn) [inline .github/workflows/styles.yml:23]: applies to all of them of course.

File: Cargo.lock [hunk 3]
```diff
@@ -1169,20 +1168,6 @@ dependencies = [
  "darling_macro 0.20.3",
 ]
 
-[[package]]
```
Comment (gnunicorn) [inline Cargo.lock:1172]: yay for de-duplicating \o/

File: Cargo.lock [hunk 4]
```diff
@@ -1700,7 +1617,7 @@ dependencies = [
 [[package]]
 name = "ffi-gen"
 version = "0.1.13"
-source = "git+https://github.com/acterglobal/ffi-gen?branch=proc-macro2-unpin#5d84ef50f30f7a26836183798611c0aa5308e455"
+source = "git+https://github.com/acterglobal/ffi-gen?branch=proc-macro2-unpin#ef009dd0e8492b9f50290ac670f1febad4390a2c"
```
Comment (gnunicorn) [inline Cargo.lock:1620]: shouldn't we start to move these to `main` on that repo?

File: rust-toolchain [hunk 5]
```diff
@@ -1,5 +1,5 @@
 [toolchain]
-channel = "nightly-2023-04-10"
+channel = "nightly-2024-01-19"
```
Comment (gnunicorn) [inline rust-toolchain:2]: here, too...
```suggestion
channel = "nightly"
```

**Review by gnunicorn — APPROVED** [review]
just the pinning, everything else looks good.

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (bitfriend) [reply .github/workflows/deploy.yml]: Fixed by https://github.com/acterglobal/a3/pull/1305/commits/969a411a23b1cea27856a14cd0a064efb70d1321

File: .github/workflows/styles.yml (hunk 2 above)
Comment (bitfriend) [reply .github/workflows/styles.yml]: Fixed by https://github.com/acterglobal/a3/pull/1305/commits/969a411a23b1cea27856a14cd0a064efb70d1321

File: rust-toolchain (hunk 5 above)
Comment (bitfriend) [reply rust-toolchain]: Fixed by https://github.com/acterglobal/a3/pull/1305/commits/969a411a23b1cea27856a14cd0a064efb70d1321

**Comment by gnunicorn** [comment]
@bitfriend question: why are we not updating to latest `matrix-sdk` on their repo here?

**Comment by bitfriend** [comment]
> @bitfriend question: why are we not updating to latest `matrix-sdk` on their repo here?

I was planning to solve it in another PR.
Do you want to solve it here?

**Pull request merged**